posts/2023/installing-farmos-on-ddev-with-composer/index.md

```diff
@@ -0,0 +1,105 @@
+ddev drush uli
+ddev launch
+```
+
+## Usage
+
+### Useful DDEV commands:
+- Start/stop a project: `ddev start`/ `ddev stop`
+- Open project: `ddev launch`
+- Stop ddev: `ddev poweroff`
+- PHP CLI: `ddev php`
+- Additional DDEV commands: https://ddev.readthedocs.io/en/latest/users/usage/cli/
+
+### Drush
+- Drush commands: `ddev drush {drush command}`
+- Login link: `ddev drush uli`
+- CRON: `ddev drush cron`
+- Additional Drush commands: https://www.drush.org/latest/commands/all/
+
+### Composer:
+- Add dependency (exmple `farm_organic`): `ddev composer require drupal/farm_organic`
+- Updating dependencies: `ddev composer update`
+- See farmOS documentation on using Composer: https://farmos.org/hosting/composer/
+
+# Going to production
+
+It should be noted that DDEV _is not_ designed to be used as a production
+hosting environment. DDEV is designed to make managing a local development
+environment easier. Taking a local environment to production depends entirely
+on how that production environment is built. Most composer-based workflows
+will expect the project's `composer.json`, `composer.lock` and any application
+code be checked in to source control, from which the production environment
+can be built.
+
+DDEV provides integrations with hosting providers to simplify the process of
+syncing production data into a local environment. For more information see
+[Hosting Provider Integration](https://ddev.readthedocs.io/en/stable/users/providers/)
+
+# Docker image limitations
+
+farmOS has been using Docker and Docker Compose to create and distribute
+reproducible development environments for the past many years. Each commit to
+farmOS builds a new Docker image that includes the environment and development
+dependencies required to install, test and lint the core codebase. While this
+process has worked well to onboard many new contributors and developers of
+farmOS, it has not been without a few headaches.
+
+One of the larger issues has simply been that installing Docker in different
+operating systems is not always a trivial process. Once installed, there are
+often additional steps required to property configure, start and use the local
+Docker environment. Slow file systems in macOS and the WSL2 layer in Windows
+are two of the most common examples that come to mind.
+
+Another issue with this process has been using the single farmOS Docker image
+to do development of contributed and custom modules (or entire project sites!)
+that are separate from farmOS core. These use cases often require a more
+complex understanding of Docker volumes and composer to properly manage and
+update project dependencies that are not compatible with the simpler
+`docker pull` update strategy the farmOS docker image is designed for. It also
+becomes more challenging to share and recreate these custom develop enviroments
+with members on a larger team.
+
+If you experience any of these issues while developing farmOS, give DDEV a try!
```

Comment:
Nit: I prefer making the call to action more broad/sweeping instead of making it predicated (if only weakly) on already having run into issues with the official Docker instructions. How about:

```suggestion
DDEV offers a refreshing alternative to Docker for developing in/on farmOS - consider trying it out!
```

Reply:
Good point. However DDEV still is docker... :-)

```suggestion
DDEV offers a refreshing experience for local development of farmOS in Docker - consider trying it out!
```